pages/calendar.js

```diff
@@ -0,0 +1,99 @@
+import { Calendar, momentLocalizer, Views } from "react-big-calendar";
+import moment from "moment";
+import LayoutStyle from "../components/LayoutStyle";
+import back from "../public/images/back.svg";
+import { useRouter } from "next/router";
+import Image from "next/image";
+import "react-big-calendar/lib/css/react-big-calendar.css";
+import styled from "styled-components";
+import { useMemo } from "react";
+import { getSession } from "next-auth/react";
+import Task from "../models/Task";
+import dbConnect from "../lib/dbConnect";
+
+export const getServerSideProps = async (context) => {
+  const session = await getSession(context);
+  if (!session) {
+    return {
+      redirect: {
+        destination: "/onboarding",
+      },
+    };
+  }
+
+  await dbConnect();
+
+  const result = await Task.find({ email: session.user.email });
+
+  const tasks = result.map((item) => {
+    const startDate = item.start;
+    const endDate = new Date(item.start.getTime() + item.totalTime * 1000);
+
+    return {
+      id: item.id,
+      title: item.name,
+      start: startDate.toJSON(),
+      end: endDate.toJSON(),
+    };
+  });
+  return { props: { tasks: tasks } };
+};
+
+export default function MyCalendar({ tasks }) {
+  const router = useRouter();
+  const localizer = momentLocalizer(moment);
+  const { defaultDate, views } = useMemo(
+    () => ({
+      defaultDate: new Date(),
+      views: {
+        day: true,
+      },
+    }),
+    []
+  );
+
+  const events = tasks.map((item) => ({
+    id: item.id,
+    title: item.title,
+    start: new Date(item.start),
+    end: new Date(item.end),
+  }));
+
+  return (
+    <LayoutStyle>
+      <ImageContainer onClick={() => router.push("/")}>
+        <Image src={back} alt="button to homepage" />
+      </ImageContainer>
+      <CalenderWrap>
+        <Calendar
+          localizer={localizer}
+          defaultDate={defaultDate}
+          events={events}
+          views={views}
+          defaultView={Views.DAY}
+          step={60}
+          min={new Date(2022, 10, 10, 6, 0)}
```

Comment:
Why do you put here a constant Date as the min?

Reply:
@F-Kirchhoff because I want the time to start at 6:00, the year, month and day here are not important because I have defined the default date, the default date will override the date 2022.10.10 I have defined in min.